packages/block-library/src/video/edit.native.js

```diff
@@ -160,6 +161,21 @@ class VideoEdit extends Component {
 		setAttributes( { id, src: url } );
 	}
 
+	onSelectURL( payload ) {
+		const { attributes, setAttributes } = this.props;
+
+		if ( payload !== attributes.src ) {
+			if ( isURL( payload ) ) {
```

Comment:
The [web implementation](https://github.com/WordPress/gutenberg/blob/4753e0ec53c37d7059caf95e5bd97b3361f5b0d4/packages/block-library/src/video/edit.js#L118-L124) first checks if the url can be converted into an embed. We probably need to do something similar in order to make sure things like youtube links work here.